src/transformers/models/mamba2/modeling_mamba2.py

```diff
@@ -359,6 +582,17 @@ def cuda_kernels_forward(
                     dim=-1,
                 )
 
+                # storing the states
+                # If we just take xBC[:, :, -self.d_conv :], it will error if seqlen < self.d_conv
```

Comment:
this was an existing generation bug. the `conv_states` were not properly stored in the first generation forward